junit-jupiter-api/src/main/java/org/junit/jupiter/api/AssertLinesMatch.java

```diff
@@ -0,0 +1,165 @@
+				return;
+			}
+		}
+
+		assertLinesMatchWithFastForward(expectedLines, actualLines);
+	}
+
+	private static void assertLinesMatchWithFastForward(List<String> expectedLines, List<String> actualLines) {
+		Deque<String> expectedDeque = new LinkedList<>(expectedLines);
+		Deque<String> actualDeque = new LinkedList<>(actualLines);
+
+		while (!expectedDeque.isEmpty()) {
+			String expectedLine = expectedDeque.pop();
+			String actualLine = actualDeque.peek();
+
+			// trivial case: take the fast path when they simply match
+			if (matches(expectedLine, actualLine)) {
+				actualDeque.pop();
+				continue;
+			}
+
+			// fast-forward marker found in expected line: fast-forward actual line...
+			if (isFastForwardLine(expectedLine)) {
+				int fastForwardLimit = parseFastForwardLimit(expectedLine);
+
+				// trivial case: fast-forward marker was in last expected line
+				if (expectedDeque.isEmpty()) {
+					int actualRemaining = actualDeque.size();
+					// no limit given or perfect match? we're done.
+					if (fastForwardLimit == Integer.MAX_VALUE || fastForwardLimit == actualRemaining) {
+						return;
+					}
+					fail(format("terminal fast-forward(%d) error: fast-forward(%d) expected", fastForwardLimit,
+						actualRemaining));
+				}
+
+				// peek next expected line
+				expectedLine = expectedDeque.peek();
+
+				// fast-forward limit was given: use it
+				if (fastForwardLimit != Integer.MAX_VALUE) {
+					// fast-forward now: actualDeque.pop(fastForwardLimit)
+					for (int i = 0; i < fastForwardLimit; i++) {
+						actualDeque.pop();
+					}
+					if (actualDeque.isEmpty()) {
+						fail(format("%d more lines expected, actual lines is empty", expectedDeque.size()));
+					}
+					continue;
+				}
+
+				// fast-forward "unlimited": until next match
+				while (true) {
+					if (actualDeque.isEmpty()) {
+						fail(format("no match for `%s` line fast-forwarding: %n%s", expectedLine, actualLines));
+					}
+					if (matches(expectedLine, actualDeque.pop())) {
+						break;
+					}
+				}
+			}
+		}
+
+		// after math
+		if (!actualDeque.isEmpty()) {
+			fail("more actual lines than expected: " + actualDeque.size());
+		}
+	}
+
+	private static boolean isFastForwardLine(String line) {
+		line = line.trim();
+		return line.startsWith(">>") && line.endsWith(">>");
+	}
+
+	private static int parseFastForwardLimit(String fastForwardLine) {
+		String text = fastForwardLine.trim().substring(2, fastForwardLine.length() - 2).trim();
+		try {
+			int limit = Integer.parseInt(text);
+			condition(limit > 0, "fast-forward must greater than zero, it is: " + limit);
+			return limit;
+		}
+		catch (NumberFormatException e) {
+			return Integer.MAX_VALUE;
+		}
+	}
+
+	private static boolean matches(String expectedLine, String actualLine) {
+		if (expectedLine.equals(actualLine)) {
+			return true;
+		}
+		try {
+			Pattern pattern = Pattern.compile(expectedLine);
+			Matcher matcher = pattern.matcher(actualLine);
+			return matcher.matches();
```

Comment:
Have you considered using the shorthand `return expectedLine.matches(actualLine);` for brevity?